pkg/apis/sources/v1beta1/apiserver_lifecycle.go

```diff
@@ -0,0 +1,139 @@
+}
+
+// GetUntypedSpec returns the spec of the ApiServerSource.
+func (s *ApiServerSource) GetUntypedSpec() interface{} {
+	return s.Spec
+}
+
+// GetCondition returns the condition currently associated with the given type, or nil.
+func (s *ApiServerSourceStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return apiserverCondSet.Manage(s).GetCondition(t)
+}
+
+// GetTopLevelCondition returns the top level condition.
+func (s *ApiServerSourceStatus) GetTopLevelCondition() *apis.Condition {
+	return apiserverCondSet.Manage(s).GetTopLevelCondition()
+}
+
+// InitializeConditions sets relevant unset conditions to Unknown state.
+func (s *ApiServerSourceStatus) InitializeConditions() {
+	apiserverCondSet.Manage(s).InitializeConditions()
+}
+
+// MarkSink sets the condition that the source has a sink configured.
+func (s *ApiServerSourceStatus) MarkSink(uri *apis.URL) {
+	s.SinkURI = uri
+	if uri != nil {
+		apiserverCondSet.Manage(s).MarkTrue(ApiServerConditionSinkProvided)
+	} else {
+		apiserverCondSet.Manage(s).MarkFalse(ApiServerConditionSinkProvided, "SinkEmpty", "Sink has resolved to empty.%s", "")
+	}
+}
+
+// MarkNoSink sets the condition that the source does not have a sink configured.
+func (s *ApiServerSourceStatus) MarkNoSink(reason, messageFormat string, messageA ...interface{}) {
+	apiserverCondSet.Manage(s).MarkFalse(ApiServerConditionSinkProvided, reason, messageFormat, messageA...)
+}
+
+// PropagateDeploymentAvailability uses the availability of the provided Deployment to determine if
+// ApiServerConditionDeployed should be marked as true or false.
+func (s *ApiServerSourceStatus) PropagateDeploymentAvailability(d *appsv1.Deployment) {
+	deploymentAvailableFound := false
+	for _, cond := range d.Status.Conditions {
+		if cond.Type == appsv1.DeploymentAvailable {
+			deploymentAvailableFound = true
+			if cond.Status == corev1.ConditionTrue {
+				apiserverCondSet.Manage(s).MarkTrue(ApiServerConditionDeployed)
+			} else if cond.Status == corev1.ConditionFalse {
+				apiserverCondSet.Manage(s).MarkFalse(ApiServerConditionDeployed, cond.Reason, cond.Message)
+			} else if cond.Status == corev1.ConditionUnknown {
+				apiserverCondSet.Manage(s).MarkUnknown(ApiServerConditionDeployed, cond.Reason, cond.Message)
+			}
+		}
+	}
+	if !deploymentAvailableFound {
+		apiserverCondSet.Manage(s).MarkUnknown(ApiServerConditionDeployed, "DeploymentUnavailable", "The Deployment '%s' is unavailable.", d.Name)
+	}
+}
+
+// MarkEventTypes sets the condition that the source has set its event type.
+func (s *ApiServerSourceStatus) MarkEventTypes() {
+	apiserverCondSet.Manage(s).MarkTrue(ApiServerConditionEventTypeProvided)
+}
+
+// MarkNoEventTypes sets the condition that the source does not its event type configured.
+func (s *ApiServerSourceStatus) MarkNoEventTypes(reason, messageFormat string, messageA ...interface{}) {
+	apiserverCondSet.Manage(s).MarkFalse(ApiServerConditionEventTypeProvided, reason, messageFormat, messageA...)
+}
```

Comment:
same here, I think nobody is calling this. We should get rid of these methods.
The reconciler used to, but I think I removed it